### DISCUSSION TITLE
valve température compensation

### DISCUSSION THREAD

**Discussion by pbranly** [jmcollin78/versatile_thermostat#175] [Q&A] [answered]
Hi
Sorry if this question had already been raised.

I use many TRVs in my water central heating.
It seems that versatile thermostat does not offer the possibility to compensate the objective temperature of each TRV depending upon the difference between the external sensor and the TRV internal sensor. 
When heating, the internal temperature sensor is heated by the radiator and the valve is too quickly closed.
Why don't you offer the possibility to compensate the objective temperature with the delta between external temperature sensor and internal one ( external is not outside !) as in the "better thermostat" ?
Is it taken into account in the versatile algorithm somewhere ?
Thanks 
Phil

**Comment by maia** [comment, marked as answer]
What you're suggesting/asking has been implemented in the past ~2 weeks, it's named "Self-regulation" in the configuration. For water based heating systems and TRVs I suggest using the "slow" option available in v4.0.0 – and please report after a few days how it's going, ideally with charts showing the room temperature, the TRV temperature, the target temperature and the "regulated temperature" (which is the target temperature plus the current offset to adjust for the discrepancy).

**Reply by pbranly** [reply]
Thanks for the answer
Do you mean the option auto-régulation-slow ?
![image](https://github.com/jmcollin78/versatile_thermostat/assets/22018509/56250e64-0e42-4385-9225-19ff12535e8a)

**Reply by maia** [reply]
yes, exactly. It seems to be missing a locale.

**Comment by jmcollin78** [comment]
> It seems to be missing a locale.

I have seen this and it is in 4.0.1 release just published with another small regression due to PR #150
